src/go/privileged-service/pkg/port/portproxy.go

```diff
@@ -51,11 +81,14 @@ func addProxy(portMapping types.PortMapping) error {
 			}
 		}
 	}
+	p.mutex.Lock()
+	p.portMappings[getHash(portProxy)] = portProxy
+	p.mutex.Unlock()
```

Comment:
Yes, this could potentially happen but remember the purpose of `portMappings` map is to remove all the port mappings upon shutdown and it ignores any potential errors from `netsh interface portproxy delete...`. So, if there is an attempt to delete an entry that doesn't have a corresponding `netsh` it's ok because the error is ignored. The most important thing is to make sure the `netsh` is up to date and that's why that operation takes precedence.